hangman.py

```diff
@@ -0,0 +1,308 @@
+    # wordlist: list of strings
+    wordlist = line.split()
+    print("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+
+def choose_word(wordlist):
+    """
+    wordlist (list): list of words (strings)
+
+    Returns a word from wordlist at random
+    """
+    return random.choice(wordlist)
+
+
+# end of helper code
+
+# -----------------------------------
+
+# Load the list of words into the variable wordlist
+# so that it can be accessed from anywhere in the program
+wordlist = load_words()
+
+
+def is_word_guessed(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    if secret_word == ''.join(letters_guessed):
+        return True
+    else:
+        return False
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    t_word = ""
+    for i in secret_word:
+        if i in letters_guessed:
+            t_word += i
+        else:
+            t_word += ("_ ")
+    return (t_word)
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    ran = string.ascii_lowercase
+    available_letters = ""
+    for i in range(len(ran)):
+        if ran[i] not in letters_guessed:
+            available_letters += ran[i]
+    return (available_letters)
+
+
+def hangman(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+
+    guesses_remaining = 6
+    warnings_remaining = 3
```

Comment:
I'd use a constants for initial values, declared in uppercase in the top of file

Reply:
But after this operation I couldn't use this values in functions.
Just catch mistake "local variable '...' referenced before assignment" which I think is understandable, so why I should declared them In the top of file?

Reply:
I don't mean you need declare these variables in the top of file. I mean - declare a constants for these digits - 6, 3 (initial values), and some others